cmake/kokkos_enable_devices.cmake

```diff
@@ -19,10 +19,17 @@ KOKKOS_CFG_DEPENDS(DEVICES NONE)
 KOKKOS_DEPRECATED_LIST(DEVICES ENABLE)
 
 
-KOKKOS_DEVICE_OPTION(PTHREAD       OFF HOST "Whether to build C++ thread backend")
-IF (KOKKOS_ENABLE_PTHREAD)
-  #patch the naming here
+KOKKOS_DEVICE_OPTION(THREADS OFF HOST "Whether to build C++ threads backend")
+IF(Kokkos_ENABLE_PTHREAD)  # for backward compatibility
+  SET(Kokkos_ENABLE_THREADS ON CACHE BOOL "Whether to build C++ threads backend")
   SET(KOKKOS_ENABLE_THREADS ON)
+  LIST(APPEND KOKKOS_ENABLED_DEVICES THREADS)
+  SET(KOKKOS_HAS_HOST ON)
+  IF(Kokkos_ENABLE_DEPRECATED_WARNINGS3)
+    MESSAGE(WARNING "The Kokkos_ENABLE_PTHREAD option is deprecated. Use Kokkos_ENABLE_THREADS instead!")
```

Comment:
We can use `MESSAGE(DEPRECATED` to implement htis functionality in a more standard way (user can pass in `-wno-deprecated` to suppress.

Reply:
Updated